crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -402,6 +405,35 @@ impl BaseClient {
             }
         }
     }
+
+    pub(crate) fn deserialize_state_events_from_timeline(
+        raw_events: &[Raw<AnySyncTimelineEvent>],
+    ) -> Vec<(Raw<AnySyncStateEvent>, AnySyncStateEvent)> {
+        raw_events
+            .iter()
+            .filter_map(|raw_event| {
+                if raw_event.get_field::<serde::de::IgnoredAny>("state_key").is_ok() {
```

Comment:
`get_field` returns `Result<Option<T>, serde_json:Error>`. It would return `Err` if the field exists but deserialization as `IgnoredAny` fails (which is impossible). You have to check whether the inner `Option` is `Some`, e.g. using `.transpose().is_some()`.

Reply:
Good catch, thanks.